src/components/calendar/calendar.js

```diff
@@ -371,13 +384,7 @@ export const BCalendar = Vue.extend({
     // Computed props that return date formatter functions
     formatDateString() {
       // Returns a date formatter function
-      return createDateFormatter(this.calendarLocale, {
-        year: 'numeric',
-        month: 'long',
-        day: 'numeric',
-        weekday: 'long',
-        calendar: 'gregory'
-      })
+      return createDateFormatter(this.calendarLocale, this.formatLong)
```

Comment:
We may want to spread in the calendar option to always ensure that the format is in the gregory calendar format (which the component only supports):

```js
return createDateFormatter(this.calendarLocale, { ...this.formatLong, calendar: 'gregory' })
```

And remove the `calendar` option from the default (not all browsers support this option, which is why we also append `u-ca-gregory` to the locale string)

Reply:
And maybe we should ensure that year, month, day should always be present:

```js
return createDateFormatter(this.calendarLocale, {
  year: 'numeric',
  month: 'numeric',
  day: 'numeric',
  ...this.formatLong,
  calendar: 'gregory'
})
```

Reply:
Personally, for my usecase I would not want to to have the `year` component displayed due to space (and it's clear due to context).

Reply:
But for screen readers, it is best to have the year, month and date formatted.  Screen reader users do not have the luxury of seeing the whole picture at once and seeing the year listed below and associating it with the value that is spoken out when the date is selected (the header is an aria-live region which announces the selected date to the blind user).